wbb/modules/greetings.py

```diff
@@ -15,66 +15,67 @@
 from wbb.utils.errors import capture_err
 from wbb.modules.admin import list_admins
-from pyrogram.types import Message, InlineKeyboardMarkup, InlineKeyboardButton
-from pyrogram.types import ChatPermissions
+from pyrogram.types import Message, InlineKeyboardMarkup, InlineKeyboardButton, ChatPermissions, User
 from pyrogram.errors.exceptions.bad_request_400 import UserNotParticipant
 
 @app.on_message(filters.new_chat_members)
 @capture_err
 async def welcome(_, message: Message):
     """Mute new member and send message with button"""
-    new_members = [f"{u.mention}" for u in message.new_chat_members]
-    text = (f"Welcome, {', '.join(new_members)}\n**Are you human?**\n"
+    for member in message.new_chat_members:
+        text = (f"Welcome, {(member.mention())}\n**Are you human?**\n"
             "You will be removed from this chat if you are not verified "
             f"in {WELCOME_DELAY_KICK_SEC} seconds")
-    if message.from_user.id in await list_admins(message.chat.id):
-        return
-    await message.chat.restrict_member(message.from_user.id, ChatPermissions())
+        await message.chat.restrict_member(member.id, ChatPermissions())
 
-    button_message = await message.reply(
-        text,
-        reply_markup=InlineKeyboardMarkup(
-            [
+        button_message = await message.reply(
+            text,
+            reply_markup=InlineKeyboardMarkup(
                 [
-                    InlineKeyboardButton(
-                        "Press Here to Verify",
-                        callback_data="pressed_button"
-                    )
+                    [
+                        InlineKeyboardButton(
+                            "Press Here to Verify",
+                            callback_data="pressed_button {}".format(member.id) # According to the docs, the user can send arbitrary data in this field, yet, I'm not able to do so, so until there's a db to pass data, imma use it like this so it will be very easy to change it to use a db in the future
+                        )
+                    ]
                 ]
-            ]
-        ),
-        quote=True
-    )
-    await message.reply_animation("CgACAgQAAx0CWIlO9AABATjjYBrlKqQID0SQ7Ey7bkVRKM5gNn8AAl8CAAIm39VRHDx6EoKU6W0eBA")
-    await kick_restricted_after_delay(WELCOME_DELAY_KICK_SEC, button_message)
-
+            ),
+            quote=True
+        )
+        #await message.reply_animation("CgACAgQAAx0CWIlO9AABATjjYBrlKqQID0SQ7Ey7bkVRKM5gNn8AAl8CAAIm39VRHDx6EoKU6W0eBA") 
+        # wtf is this? "file_id is unique for each individual bot and can't be transferred from one bot to another." source: https://core.telegram.org/bots/api#sending-files
+        #and here is the error I just received -- pyrogram.errors.exceptions.bad_request_400.MediaEmpty: [400 MEDIA_EMPTY]: The media you tried to send is invalid (caused by "messages.SendMedia")
+        asyncio.create_task(kick_restricted_after_delay(WELCOME_DELAY_KICK_SEC, button_message, member))
```

Comment:
I think that just storing it in memory and killing it when the user verifies himself will already improve the performance